cli/src/commands/git/clone.rs

```diff
@@ -287,24 +307,34 @@ fn fetch_new_remote(
             },
         )
     })?;
+    let specified_target_branch = target_branch.is_some();
     let default_branch = git_fetch.get_default_branch(remote_name)?;
+    let fetched_branch = target_branch.or(default_branch.clone());
+    let fetched_default = fetched_branch == default_branch;
     let import_stats = git_fetch.import_refs()?;
-    if let Some(name) = &default_branch {
+    if let Some(name) = &fetched_branch {
         let remote_symbol = name.to_remote_symbol(remote_name);
         let remote_ref = tx.repo().get_remote_bookmark(remote_symbol);
-        if track_default && remote_ref.is_present() {
+
+        if specified_target_branch && !remote_ref.is_present() {
+            return Err(user_error(format!(
+                "No such branch: {}",
+                name.clone().into_string()
+            )));
+        }
+        if (fetched_default && should_track_default) || git_settings.auto_local_bookmark {
```

Comment:
Why `|| git_settings.auto_local_bookmark`?

Reply:
It’s my understanding that when auto_local_bookmark is true, all fetched branches are tracked, so we should apply that here

Reply:
I think `auto_local_branch` should be handled by the jj-lib code. Can you add a separate commit with test if you found a bug in the current implementation?

Reply:
You are right, it is already handled.